COMPONENT-GUIDELINES.md

```diff
@@ -0,0 +1,293 @@
+# Purpose
+
+This document defines guidelines and patterns developers should follow when building components for the Pipedream registry.
+
+Developers may create, deploy and share components that do not conform to these guidelines, but they will not be eligible to be listed in the curated registry (e.g., they may be hosted in a Github repo). If you develop a component that does not adhere to these guidelines, but you believe there is value to the broader community, please [reach out in our community forum](https://pipedream.com/community). We can either provide guidance to improve compliance or evaluate if an exception is warranted.
+
+# Overview
+
+[Pipedream](https://pipedream.com) is a low code integration platform that makes it easy to connect APIs remarkably fast. Users can select from thousands of customizable, open source components for hundreds of apps, and orchestrate their execution in workflows. Developers can [contribute](CONTRIBUTING.md) to these open source components on [Github](https://github.com/pipedreamhq/pipedream) by:
```

Comment:
Grammar nitpick: The comma in this phrase can be eliminated: "open source components for hundreds of apps and orchestrate their execution in workflows." or changed to "open source components for hundreds of apps, and they can orchestrate their execution in workflows."